src/main/scala/com/wavesplatform/matcher/model/OrderValidator.scala

```diff
@@ -106,7 +109,13 @@ object OrderValidator {
       exchangeTx.flatMap(verifySmartToken(blockchain, assetId, _))
     }
 
-    lazy val mof = ExchangeTransactionCreator.minFee(blockchain, matcherAddress, order.assetPair)
+    val baseFee = orderFeeSettings match {
+      case FixedWavesSettings(fee) => fee
+      case _                       => exchangeTransactionCreationFee
+    }
```

Comment:
Actually, this isn't the true for `case _`. I agree, this case isn't real, but IMO it'd better to do something like:
```scala
def mof(baseFee: Long) = ExchangeTransactionCreator.minFee(blockchain, matcherAddress, order.assetPair, baseFee)
// ...
        .ensure(MatcherError.FeeNotEnough(mof, order.matcherFee, None)) { o =>
          orderFeeSettings match {
            case FixedWavesSettings(baseFee) => o.matcherFee >= mof(baseFee)
            case _                           => true
          }
        }
```

Reply:
Slightly better way:
```scala
for {
// …
_ <- orderFeeSettings match {
  case FixedWavesSettings(baseFee) =>
    val mof = ExchangeTransactionCreator.minFee(blockchain, matcherAddress, order.assetPair, baseFee)
    Either.cond(o.matcherFee >= mof, success, MatcherError.FeeNotEnough(mof, order.matcherFee, None))
  case _                           => success
}
// …
]
```

Reply:
Slightly better way:
```scala
for {
// …
_ <- orderFeeSettings match {
  case FixedWavesSettings(baseFee) =>
    val mof = ExchangeTransactionCreator.minFee(blockchain, matcherAddress, order.assetPair, baseFee)
    Either.cond(o.matcherFee >= mof, success, MatcherError.FeeNotEnough(mof, order.matcherFee, None))
  case _                           => success
}
// …
]
```

Reply:
Done